### ISSUE TITLE
Is the Wishbone in itself licensed/protected in any way?

### ISSUE THREAD

**Issue by m-kru** [fossi-foundation/wishbone#24]
The Wishbone B3 specification doesn't have any copyright, Wishbone B4 has copyright. However, what about the whole Wishbone idea and the design concept? Is it protected by any license or patent? I am wondering how much can one be inspired by Wishbone when designing his own bus.

**Comment by wallento** [comment]
The B4 spec has been highly disputed because of that (along technical issues). I would recommend starting from B3, which also was the initial goal when we tried to support the standard. Unfortunately, the steward @rherveille didn't further participate in the dicussions and the entire project is stalled since then.

Starting from the B3 spec was the initial goal and some work was done, we are happy to discuss proceeding under a different name if you are planning to permanently evolve it.

**Comment by m-kru** [comment]
Is there any mailing list?

**Comment by rherveille** [comment]
Sorry, I am very busy with projects.
What do we need to do and what are you awaiting from me?

Richard

Sent from my iPhone

On 30 Mar 2022, at 10:09, Stefan Wallentowitz ***@***.***> wrote:

﻿

The B4 spec has been highly disputed because of that (along technical issues). I would recommend starting from B3, which also was the initial goal when we tried to support the standard. Unfortunately, the steward @rherveille<https://github.com/rherveille> didn't further participate in the dicussions and the entire project is stalled since then.

Starting from the B3 spec was the initial goal and some work was done, we are happy to discuss proceeding under a different name if you are planning to permanently evolve it.

**Comment by m-kru** [comment]
@rherveille personally I don't want anything. I am wondering how much I can take from the Wishbone when designing similar bus.

**Comment by rherveille** [comment]
Besides the obvious question; why do you want to design a similar bus instead of using Wishbone?
The Wishbone spec is placed in the public domain. The document, as is, can be used without restrictions.
However ‘public domain’ does not mean free of copyright or other rights, especially given the large variety of legislation in the world around those subjects.
Also keep in mind that for Wishbone a large patent investigation was done. Wishbone almost certainly doesn’t infringe any patents, but you never know until it’s actually gone to court. Having said that, given that wishbone almost celebrates its 25anniversary, I think it is safe to say it’s safe.
However any derivate you develop might infringe on someone’s patents or copyrights.

Richard


From: m-kru ***@***.***>
Date: Wednesday, 30 March 2022 at 21:41
To: fossi-foundation/wishbone ***@***.***>
Cc: Richard Herveille ***@***.***>, Mention ***@***.***>
Subject: Re: [fossi-foundation/wishbone] Is the Wishbone in itself licensed/protected in any way? (Issue #24)

@rherveille<https://github.com/rherveille> personally I don't want anything. I am wondering how much I can take from the Wishbone when designing similar bus.

**Comment by m-kru** [comment]
> Besides the obvious question; why do you want to design a similar bus instead of using Wishbone?

Because I don't like what happened with Wishbone in B4. The specification is no longer orthogonal. Standard cycle with registered feedback bus cycle duplicates the functionality of pipelined cycle capabilities. Generally the pipelined mode is more robust and generic and this is the only advantage of B4. I think the standard mode should be removed. I don't like the fact that error is single bit information, this is not enough and extending it with data tag is not a solution that scales well. I also think that operand size and granularity should not be part of the bus specification. Maybe granularity, but for sure not the operand size.

**Comment by wallento** [comment]
Fully agree. We hat this discussion and therefore started a new spec from B3. Unfortunately it was stalled due to lack of time and responses. We can release an update from the current master branch if @rherveille agrees. We would welcome any further development of B3 towards further revisions. @rherveille is the steward here, maybe you can delegate the further development to interested developers that have the bandwidth?

**Comment by m-kru** [comment]
@rherveille ping

**Comment by rherveille** [comment]
:) It's probably time to start again

**Comment by ams** [comment]
> The Wishbone spec is placed in the public domain. The document, as is, can be used without restrictions. However ‘public domain’ does not mean free of copyright or other rights, especially given the large variety of legislation in the world around those subjects.

That is not the case. PD menas that it is indeed without restrictions and free of copyright. The vast legislation in the world is based on the Berne convention.

The real question is if WB is infact in the PD, how a work is put into the PD very specific in each country, and in some you cannot do so (other than dying and waiting many years).

**Comment by rherveille** [comment]
*   That is not the case. PD menas that it is indeed without restrictions and free of copyright. The vast legislation in the world is based on the Berne convention.
  *   The real question is if WB is infact in the PD, how a work is put into the PD very specific in each country, and in some you cannot do so (other than dying and waiting many years).
That is not the case. In some countries you can not relinquish copyright. Meaning a document is always covered under some copyright and owned by some entities.
However Wishbone is consider to be the public domain as its understood by most people and legislations. The document and its contents can be used and copied without restrictions. Each individual contributor gave up any rights to their respective works in writing.
Wishbone B4 veered of that path and specifically added copyright (which goes against the Wishbone principle and the wish of its original inventor). Hence I consider Wishbone B4 not part of the official B4 spec, but a branch.

Richard


From: Alfred M. Szmidt ***@***.***>
Date: Saturday, 20 May 2023 at 01:56
To: fossi-foundation/wishbone ***@***.***>
Cc: Richard Herveille ***@***.***>, Mention ***@***.***>
Subject: Re: [fossi-foundation/wishbone] Is the Wishbone in itself licensed/protected in any way? (Issue #24)

The Wishbone spec is placed in the public domain. The document, as is, can be used without restrictions. However ‘public domain’ does not mean free of copyright or other rights, especially given the large variety of legislation in the world around those subjects.

That is not the case. PD menas that it is indeed without restrictions and free of copyright. The vast legislation in the world is based on the Berne convention.

The real question is if WB is infact in the PD, how a work is put into the PD very specific in each country, and in some you cannot do so (other than dying and waiting many years).

**Comment by ams** [comment]
No, it is absolutley the case.  PD literally means copyright does not
apply.

Copyright "was added" as soon as the work was created, you literally
cannot relinquish copyright in any country.  It is not how things
work.  If that is or isn't the intent of the document writers I won't
speculate, but having dabbled in this quagmire for several decades, it
pains me to see that this confusion still perpetuates.

In some sense, B4 is beter from a legal point of view in that it
actually has a clear copyright status, while B3 does not.  For the
same reasons why many projects reject "my changes are in the PD' and
instead opt that one explicitly licenses the work under a CC0 like
license (which has equivalent properties as PD -- but still copyrighted).


If the intent of the authors is something like PD, then I would
strongly suggest picking the CC0, which achives the exact same thing,
and doesn't go into the mess that is copyright.

**Comment by rherveille** [comment]
Time to read up on CC0 I guess.
When Wishbone was released into the public domain, we had lawyers looking into it.
Their conclusion at the time (almost 25years ago) was that Public Domain was sufficient.
But things do change over time.

Richard

From: Alfred M. Szmidt ***@***.***>
Date: Saturday, 20 May 2023 at 08:19
To: fossi-foundation/wishbone ***@***.***>
Cc: Richard Herveille ***@***.***>, Mention ***@***.***>
Subject: Re: [fossi-foundation/wishbone] Is the Wishbone in itself licensed/protected in any way? (Issue #24)
No, it is absolutley the case. PD literally means copyright does not
apply.

Copyright "was added" as soon as the work was created, you literally
cannot relinquish copyright in any country. It is not how things
work. If that is or isn't the intent of the document writers I won't
speculate, but having dabbled in this quagmire for several decades, it
pains me to see that this confusion still perpetuates.

In some sense, B4 is beter from a legal point of view in that it
actually has a clear copyright status, while B3 does not. For the
same reasons why many projects reject "my changes are in the PD' and
instead opt that one explicitly licenses the work under a CC0 like
license (which has equivalent properties as PD -- but still copyrighted).


If the intent of the authors is something like PD, then I would
strongly suggest picking the CC0, which achives the exact same thing,
and doesn't go into the mess that is copyright.

**Comment by KubaO** [comment]
> Is the Wishbone in itself licensed/protected in any way?

Yes. And that is a **good thing**. I think some of the confusion stems from conflating the *document itself* with *the ideas it presents*.

Licensing and protection by law don't imply that you have to pay for it. In case of most open source software, it prevents the author and the community from getting screwed in fact.

Thus:

1. It is perfectly fine for Wishbone-the-specification to be protected by copyright. In fact, at least in the US, it is by default protected by copyright under the all-rights-reserved terms, i.e. no license exists for anyone else to use it or do anything with it. Thus, the specification itself should be permissively licensed. It in fact must be permissively licensed, else no one can legally even access it (!).

2. Wishbone-the-idea is in public domain. That is, unless someone else nefariously would assert patent protection for it. That's why there's the prior art section in the spec, to discourage ~pond scum~ bad actors from filing baseless patents for it.

These two are entirely orthogonal.

> doesn't go into the mess that is copyright.

That doesn't make sense. The only reason there is anything to license is that there is protection by copyright law. Otherwise a license would be non-binding and anyone could do whatever they wanted.

CC0 is a fine license. It is only enforceable because if anyone breaks its terms, they lose the right to use, distribute, and make derivative works of the document. "Copyright" is not a mess. It is what allows licenses to exist and have meaning in the first place.

> In some sense, B4 is better from a legal point of view in that it actually has a clear copyright status

Not having a clear licensing status means that all rights are reserved to the authors, and the document is legally inaccessible even if it is de-facto accessible.

So, B4 is not "better". It is, by law, the first revision that can actually be used without a legal department of one's employer raising objections.

> you literally cannot relinquish copyright in any country

Well, in the US at least you can. That what "placing in public domain" actually is. But US is not the entire world, and thus placing something in public domain may have no meaning elsewhere in the world. Thus, for documents with global scope, it is *better* to use a permissive license like CC0 instead of placing the document in public domain.

---

This whole comment - and every comment in this issue - are by default protected by US copyright law. I'm not a lawyer, and am unsure how much (if any) case law there is in this regard, but typically everything that "goes with" an open-source project is licensed the same as the open source project, or under other terms that are expressly disclosed. That's why some OSS projects choose to apply a different license to the code than one applied to the documentation, issues, forums/groups etc.

Without a license, nobody but the rights holder - by default the author - has any rights whatsoever to the content other than as given by fair use provisions and corresponding clauses in laws of other countries.

**Comment by rherveille** [comment]
Since Wishbone B2 the document has been released into the public domain. And that has always been the intend of the original creators/authors. Any document copyrighting Wishbone is in violation of that initial idea.

**Comment by ams** [comment]
> CC0 is a fine license. It is only enforceable because if anyone breaks its terms, they lose the right to use, distribute, and make derivative works of the document. "Copyright" is not a mess. It is what allows licenses to exist and have meaning in the first place.
> 

That is not the case with the CC0 -- it does not have a termination clause (there are other Free Software licenses that do, e.g. the GNU GPL).   

>> In some sense, B4 is better from a legal point of view in that it actually has a clear copyright status

> Not having a clear licensing status means that all rights are reserved to the authors, and the document is legally inaccessible even if it is de-facto accessible.

The license situation with B4 is quite clear, the issue is the license it self.  B3 etc, do not have a license -- even if the intent of @rherveille is PD the Wishbone spec lacks a license or clear terms.  That is what is not deal.

> So, B4 is not "better". It is, by law, the first revision that can actually be used without a legal department of one's employer raising objections.

Did you forget a negation there?  A document that can be used without legal getting involved .. is a good thing, since it has a clear license situation.

> Well, in the US at least you can. That what "placing in public domain" actually is. But US is not the entire world, and thus placing something in public domain may have no meaning elsewhere in the world. Thus, for documents with global scope, it is better to use a permissive license like CC0 instead of placing the document in public domain.

In the USA there is no means to "relinquish" copyright to the PD (e.g, other than letting copyright expire by death, and waiting 90 years), or having the work published before a specific date (ca. 1977).  The issue is that the copyright author still has the means to cancel the the exclusive or nonexclusive grant they made.
